src/odata-to-abstract-sql.ts

```diff
@@ -575,34 +595,96 @@ export class OData2AbstractSQL {
 		method: string,
 		path: ODataQuery,
 		resource: Resource,
-		referencedField: ReferencedFieldNode,
 		bodyKeys: string[],
 	): BooleanTypeNodes | void {
-		if (path.key != null) {
+		const { key } = path;
+		if (key != null) {
 			if (method === 'PUT' || method === 'PUT-INSERT' || method === 'POST') {
-				// Add the id field value to the body if it doesn't already exist and we're doing an INSERT or a REPLACE.
-				const qualifiedIDField = resource.resourceName + '.' + resource.idField;
-				if (
-					!bodyKeys.includes(qualifiedIDField) &&
-					!bodyKeys.includes(resource.idField)
-				) {
-					bodyKeys.push(qualifiedIDField);
-					this.extraBodyVars[qualifiedIDField] = path.key;
+				if (isBindReference(key)) {
+					addBodyKey(
+						resource.resourceName,
+						resource.idField,
+						key,
+						bodyKeys,
+						this.extraBodyVars,
+					);
+				} else {
+					for (const [fieldName, bind] of Object.entries(key)) {
+						addBodyKey(
+							resource.resourceName,
+							fieldName,
+							bind,
+							bodyKeys,
+							this.extraBodyVars,
+						);
+					}
 				}
 			}
-			for (const matcher of [this.Bind, this.NumberMatch, this.TextMatch]) {
-				const key = matcher.call(this, path.key, true);
-				if (key) {
-					return [comparison.eq, referencedField, key];
+			if (isBindReference(key)) {
+				const bind = this.Bind(key);
+				const referencedField: ReferencedFieldNode = [
+					'ReferencedField',
+					resource.tableAlias!,
```

Comment:
Would love to see a comment why force-unwrapping here is alright.

Reply:
Actually I take it back, I can't replace it because `this.ReferencedField(...)` expects the odata name (it's a helper to handle any necessary conversion stuff) and here we use the sql name directly

Reply:
Let's put the gist of these 2 comments in the code, for reference?

Reply:
Turns out I had done the typing already in https://github.com/balena-io-modules/odata-to-abstract-sql/pull/64 but not applied it to every `.tableAlias!` usage. I've applied it everywhere and it seems to be happy :)